### DISCUSSION TITLE
Bot ideas!

### DISCUSSION THREAD

**Discussion by josefaidt** [aws-amplify/discord-bot#1] [Ideas]
Hey yall :wave: starting this thread to continue the discussion from Discord regarding bots to improve and sustain the community experience! Ideas can range from more general purpose functionalities, driving interactions and conversations, quality of life enhancements, etc. Here are a couple ideas I've gathered to get started:

- administration functionality to add roles (e.g. `/giverole staff @josef` 
- broadcast events (office hours, w/ friends, etc.) (cc `@mo`)
- auto reply to messages with text like "need help" in not-so-relevant channels such as #lobby with a message "be sure to check out our *-help channels!"
- schedule a time to allow @everyone to post in a text channel #Amplify-Office-Hours, lock messaging permissions after office hours ends, but always allow reactions and others to read message history
  - enables folks who don't feel comfortable speaking to ask a question via text
  - e.g. `/officehours open` and `/officehours close`
- [cross-post detection](https://discord.com/channels/705853757799399426/705908850305531975/850701207630905415)

Let us know what you think and post any ideas you have!

**Comment by renegoretzka** [comment]
In the future, we can also create a bot that automatically creates threads.
This would be very useful to keep questions and answers in one place. It allows us to avoid lost / unseen questions.

For more about threads and its api, read here: https://github.com/discord/discord-api-docs/pull/2855

**Reply by nubpro** [reply]
O wow, threads is coming to DISCORD!??

Once thread is introduced, setup a bot to scrap questions asked by the community and stores the answers in a QnA format on a site. Then, let Google SEO do the hardwork.

Result:
Community can now search through Discord's messages through Google :)
Basically, 1 search for everything including Discord + GitHub + Stackoverflow + ❤️

**Reply by renegoretzka** [reply]
Awesome Chai! 👍
Thank you for your feedback!

**Reply by siegerts** [reply]
re: store results on site - Create some form of living document from these (i.e GitHub Discussion :) in the correct repo to allow for more comments/answering.

**Reply by billygerhard** [reply]
The [Electron Discord](https://discord.com/invite/electronjs) has automatic threads, and its great to keep tickets in line, but it seems to limit who helps. They have a ping system that pings a role when it gets created, and the list of users when you enter a thread is limited. If we did do auto threads, we needs to make sure it doesn't limit the community from seeing issues and helping.

**Comment by n123456789k** [comment]
We should look into the typescript discord how they do help channels also that would be a good concept

**Comment by josefaidt** [comment]
from Discord user `Matt` 

> does discord have some kind of points system, e.g. gifting points to people when they help out? could be a way to encourage more people to participate in answering questions... and then if you've got a lot of points, it might help you get answers from other people when you're asking questions

https://discord.com/channels/705853757799399426/705912836148166677/856745967294742599
discord://discord.com/channels/705853757799399426/705912836148166677/856745967294742599

**Reply by alexkates** [reply]
Love this idea. I'm envisioning something like [HeyTaco](https://slack.com/apps/A0J4UNFLN-heytaco?tab=more_info) where a Discord user can run a command and give another user recognition.

Something like /amplifybot recognize @user

**Comment by billygerhard** [comment]
Have the bot record the weekly Office Hours, then it can be archived and published, maybe as a podcast for people who couldn't attend to listen to.

**Comment by Fyleek** [comment]
A discord bot that could synthesize a text like a tree of dots, to simplify navigation through the content of offices hours or for other topics.
I don't know if my idea is really clear 😅

**Reply by josefaidt** [reply]
sort of like an unordered list of popular topics, office hour topics, etc.? Maybe like a weekly/monthly digest of Amplify topics that were discussed?

**Reply by Fyleek** [reply]
Yes more like a weekly/monthly digest of Amplify topics 👍🏼

**Comment by josefaidt** [comment]
With the features of searching docs pages and GitHub issues, would it make sense to have a global `/search` command to accomplish this or keep the functionality with their respective commands such as `/docs "my query"` and `/github issues "my query"`? 

NOTE: Keeping the functionality with the respective command may exceed supported subcommand depth

Reference:

- https://github.com/aws-amplify/discord-bot/issues/94
- https://github.com/aws-amplify/discord-bot/issues/90
